queries.sql

```diff
@@ -7,4 +7,51 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name NOT IN ('Gabumon');
-SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+ 
+BEGIN;
+UPDATE animals
+SET species = 'unspecified';
+SELECT name, species FROM animals;
+ROLLBACK;
+
+BEGIN;
+UPDATE animals
+SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals 
+SET species = 'pokemon' WHERE species IS NULL;
+SELECT name, species FROM animals;
+COMMIT;
+
+-- delete and rollback
+BEGIN;
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK;
+
+
+-- delete and savepoint
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT before_deletion;
+SELECT * FROM animals;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO before_deletion;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+SELECT * FROM animals;
+COMMIT;
```

Comment:
- [ ] Kindly add the full data for the last query, I got the correct answer from your implementation. Kindly update it in order to show the correctness of your DB. ℹ️ 👇 

## Your implementation in VScode
<img width="504" alt="Untitled" src="https://github.com/Mofuhidy/vet-clinic-db/assets/103386183/9475898f-c5e7-4fe5-bd34-77c2c3e110de">

## The screenshot you uploaded
<img width="470" alt="Untitled" src="https://github.com/Mofuhidy/vet-clinic-db/assets/103386183/8500e879-b634-48cc-a937-c4a0934429ea">